packages/streaming-image-volume-loader/src/BaseStreamingImageVolume.ts

```diff
@@ -351,6 +241,17 @@ export default class BaseStreamingImageVolume
     // it is either cachedImage or cachedVolume
     const isFromImageCache = !!cachedImage;
 
+    if (isFromImageCache) {
```

Comment:
Can you check the target buffer too?  If that isn't set, you shouldn't be doing anything here.